data/src/split.py

```diff
@@ -44,7 +44,84 @@ def _detect_best_script_name(
         # The script names in Unicode data tables have underscores instead of
         # whitespace to enable parsing. See:
         #   https://www.unicode.org/Public/13.0.0/ucd/Scripts.txt
-        return script_probs[0][0].replace("_", " "), script_probs[0][1]
+        return script_probs[0][0], script_probs[0][1]
+
+
+def _remove_duplicates(script_dict: Dict[str, str]) -> Dict[str, str]:
+    '''
+    If a values in lang["script"] appears more than once, the [key:value] pair that does not conform to ISO unicode
+    entries returned from unicodedataplus.property_value_aliases['script'] is deleted.
+
+    eg)
+
+    "aze": {
+        "iso639_name": "Azerbaijani",
+        "wiktionary_name": "Azerbaijani",
+        "wiktionary_code": "az",
+        "casefold": true,
+        "script": {
+            "latn": "Latin",
+            "cyrl": "Cyrillic",
+            "ara": "Arabic",
+            "arab": "Arabic"
+        }
+
+        ->
+
+        "aze": {
+        "iso639_name": "Azerbaijani",
+        "wiktionary_name": "Azerbaijani",
+        "wiktionary_code": "az",
+        "casefold": true,
+        "script": {
+            "latn": "Latin",
+            "cyrl": "Cyrillic",
+            "arab": "Arabic"
+        }
+
+
+    '''
+    remove = []
+
+    for key, value in script_dict["script"].items():
+        value = value.replace(" ", "_")
+        if not ''.join(unicodedataplus.property_value_aliases['script'][value]).lower() == key:
+            remove.append(key)
+    for i in remove:
+        del script_dict["script"][i]
+    return script_dict
+
+
+def _update_languages_json(tsv_path: str, LANGUAGES_PATH: str) -> None:
+    '''
+    Detects and identifies all unicode scripts present in a tsv file
+    and updates languages.json to reflect updated ["script"]
+    entries for each language in languages.json
+    '''
+    with open(LANGUAGES_PATH, "r", encoding="utf-8") as lang_source:
+        languages = json.load(lang_source)
+        for file in os.listdir(tsv_path):
```

Comment:
I fear this might make `postprocess` (where `split.py` is called) take an unreasonably long time.

`postprocess` itself is already running through all the files in the `tsv` directory and calling `split.py` on each. Given the line above, `postprocess` is going to be running through every file in `tsv` and for each file, `split.py` is going to run through every file in `tsv`. Scary stuff.

I think we should either update `languages.json` before `postprocess` runs through `tsv` (meaning pull this functionality out of `split.py` and put it into a separate file which is run at the top of `postprocess`), or just update this function to rewrite `languages.json` one language at a time as each language is passed to `split.py`.

Reply:
I think updating the languages.json before postprocess makes the most sense here. `languages_update.py` runs fairly quickly on its own (no longer than a minute or two or my machine, but I can log it), and it seems more "tidy" approach to me, but what do I know? lol.

Reply:
Having not thought heavily about it I think if it runs reasonably fast (minutes, let's say) the design is solid.